cabal-install/Distribution/Client/IndexUtils.hs

```diff
@@ -262,7 +262,7 @@ getSourcePackagesAtIndexState verbosity repoCtxt mb_idxState = do
             return ()
         IndexStateTime ts0 -> do
             when (isiMaxTime isi /= ts0) $
-                warn verbosity ("Requested index-state " ++ display ts0
+                info verbosity ("Requested index-state " ++ display ts0
```

Comment:
So, my mental model of `--index-state` is that you are saying, "Give me the state of the index at time X." In this case, the date I pass in might not correspond to any timestamp explicitly mentioned in the index. But this doesn't feel like a warning case to me: the time of last update might be different from the time I gave, but what's wrong with that?